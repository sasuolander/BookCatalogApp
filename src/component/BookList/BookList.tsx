import React from 'react'
import {BookTile} from '../BookTile/BookTile'

export const BookList =({})=>{
    const array =[1,2,3,4,5,6,7,8,9,10]
    return (
   
<div>
{array.map((id)=>{return<BookTile id={id}/>})}
</div>)}
