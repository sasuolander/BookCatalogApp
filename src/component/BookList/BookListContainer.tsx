import { BookList } from "./BookList";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { ListOfBookState } from "../../model/StateOfApp";

export const BookListContainer = () => {
  interface StateFromProps extends ListOfBookState {}
  interface mapDispatchToProps {}

  const mapStateToProps = (state: any) => ({
    books: state.mainlist.books
  });

  const mapDispatchToProps = (dispatch: Dispatch) => ({});
  return connect<StateFromProps, mapDispatchToProps>(
    mapStateToProps,
    mapDispatchToProps
    // @ts-ignore
  )(BookList);
};
