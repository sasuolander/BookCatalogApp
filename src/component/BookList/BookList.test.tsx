import React from 'react'
import { BookList } from "./BookList";
import { mount, shallow, render } from "enzyme";

describe("Book list",()=>{
    it("render component Book list",()=>{
        const component = shallow(<BookList/>)
        expect(component).toMatchSnapshot();
    })
})

