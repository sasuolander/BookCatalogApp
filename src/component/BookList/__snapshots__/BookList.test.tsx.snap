// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book list render component Book list 1`] = `
<div>
  <BookTile
    id={1}
  />
  <BookTile
    id={2}
  />
  <BookTile
    id={3}
  />
  <BookTile
    id={4}
  />
  <BookTile
    id={5}
  />
  <BookTile
    id={6}
  />
  <BookTile
    id={7}
  />
  <BookTile
    id={8}
  />
  <BookTile
    id={9}
  />
  <BookTile
    id={10}
  />
</div>
`;
